src/file.cr

```diff
@@ -44,20 +70,20 @@ class File < IO::FileDescriptor
   # File.stat("foo").mtime # => 2015-09-23 06:24:19 UTC
   # ```
   def self.stat(path) : Stat
-    Crystal::System::File.stat(path)
+    stat?(path) || raise Errno.new("Unable to get stat for #{path.inspect}")
```

Comment:
Should the error message be exactly the same as for `Errno::ENOENT`?

Reply:
I thought about this, yes it should be since it gets appended the error text for `ENOENT` anyway, which is "No such file or directory".